tests/ui/manual_as_slice.fixed

```diff
@@ -0,0 +1,37 @@
+#![warn(clippy::manual_as_slice)]
+
+mod nested_1 {
+    pub(crate) mod nested_2 {
+        pub(crate) const SOME_VALUE: [u8; 4] = [1, 2, 3, 4];
+    }
+}
+
+fn main() {
+    let array: [u8; 4] = [0; 4];
+    let slice: &[u8] = array.as_slice();
+    //~^ manual_as_slice
+
+    let mut array: [u8; 4] = [0; 4];
+    let mut slice: &[u8] = array.as_mut_slice();
+    //~^ manual_as_slice
+
+    let slice: &[u8] = nested_1::nested_2::SOME_VALUE.as_slice();
+    //~^ manual_as_slice
+
+    let slice = b"foo";
+    let slice: &[u8] = slice.as_slice();
+    //~^ manual_as_slice
+
+    let slice = &[1, 2, 3, 4];
+    let slice: &[u8] = slice.as_slice();
+    //~^ manual_as_slice
+
+    macro_rules! perform_the_slice {
+        ($a:expr) => {
+            [1, 2, 3].as_slice()
```

Comment:
This is not OK. The lint has changed behavior. You can see this if you call `perform_the_slice!([42])` after line 36.

A solution to this problem is to use a MultiSpan suggestion. To do that, the suggestion in the lint can be a `Vec<(Span, String)>`, so you can build a vec of two elements, the first being the span of the ampersand with an empty String, and the second being the span of the `[..]` with `".as_slice()"`. We should however be sure that both of those come from the same expansion context before making the suggestion, otherwise we may end up at the same problem.